src/scripts/cronjobs/monthlyActivityFree.tsx

```diff
@@ -0,0 +1,151 @@
+    );
+  }
+  const subscribersToEmail = (await getFreeSubscribersWaitingForMonthlyEmail())
+    .filter((subscriber) => {
+      const assumedCountryCode = getSignupLocaleCountry(subscriber);
+      return assumedCountryCode === "us";
+    })
+    .slice(0, batchSize);
+  await initEmail();
+
+  await Promise.allSettled(
+    subscribersToEmail.map((subscriber) =>
+      sendMonthlyActivityEmail(subscriber),
+    ),
+  );
+  console.log(
+    `[${new Date(Date.now()).toISOString()}] Sent [${subscribersToEmail.length}] monthly activity emails to free users.`,
+  );
+}
+
+async function sendMonthlyActivityEmail(subscriber: SubscriberRow) {
+  const sanitizedSubscriber = sanitizeSubscriberRow(subscriber);
+  const l10n = getCronjobL10n(sanitizedSubscriber);
+  /**
+   * Without an active user session, we don't know the user's country. This is
+   * our best guess based on their locale. At the time of writing, it's only
+   * used to determine whether to count SSN breaches (which we don't have
+   * recommendations for outside the US).
+   */
+  const countryCodeGuess = getSignupLocaleCountry(subscriber);
+
+  // OneRep suggested not relying on webhooks, but instead to fetch the latest
+  // data from their API. Thus, let's refresh the data in our DB in real-time:
+  if (subscriber.onerep_profile_id !== null) {
+    await refreshStoredScanResults(subscriber.onerep_profile_id);
+  }
+
+  const latestScan = await getLatestOnerepScanResults(
+    subscriber.onerep_profile_id,
+  );
+  const subscriberBreaches = await getSubscriberBreaches({
+    fxaUid: subscriber.fxa_uid,
+    countryCode: countryCodeGuess,
+  });
+  const data = getDashboardSummary(latestScan.results, subscriberBreaches);
+
+  const subject = l10n.getString("email-monthly-free-subject");
+  const unsubscribeLink =
+    await getMonthlyActivityFreeUnsubscribeLink(subscriber);
+
+  if (unsubscribeLink === null) {
+    throw new Error(
+      `Trying to send a monthly activity email to a free user, but the unsubscribe link could not be generated: [${unsubscribeLink}].`,
+    );
+  }
+
+  // Update the last-sent date *first*, so that if something goes wrong, we
+  // don't keep resending the email a brazillion times.
+  await updateEmailPreferenceForSubscriber(subscriber.id, true, {
+    monthly_monitor_report_free_at: new Date(Date.now()),
+  });
+
+  try {
+    await sendEmail(
+      sanitizedSubscriber.primary_email,
+      subject,
+      renderEmail(
+        <MonthlyActivityFreeEmail
+          subscriber={sanitizedSubscriber}
+          dataSummary={data}
+          l10n={l10n}
+          unsubscribeLink={unsubscribeLink}
+        />,
+      ),
+    );
+  } catch (e) {
+    console.error("send_monthly_activity_email_free", {
+      exception: e,
+    });
+  }
+}
+
+export async function getMonthlyActivityFreeUnsubscribeLink(
+  subscriber: SubscriberRow,
+) {
+  try {
+    const newUnsubToken = randomBytes(64).toString("hex");
```

Comment:
Hmm, I inlined it because it was just a single line and only used here, but I see you've created and merged https://github.com/mozilla/blurts-server/pull/5149 since. Isn't that column supposed to be nullable - i.e. why do we even need to provide a default value there?

Reply:
I guess for a more specific question: do you think there's any risk in dc0ac3201?

Reply:
@Vinnl I think after you resolve the conflict with main, you will see that this code has been changed

It will always generate a token instead of "" or null?

Reply:
@mansaj Right, but I'm asking if that's necessary to do? Or if we can just keep the token-generation limited to the email-sending code where I have it now?